src/main/java/org/datadog/jenkins/plugins/datadog/apm/JavascriptConfigurator.java

```diff
@@ -0,0 +1,40 @@
+package org.datadog.jenkins.plugins.datadog.apm;
+
+import hudson.FilePath;
+import hudson.model.Node;
+import hudson.model.TaskListener;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+final class JavascriptConfigurator implements TracerConfigurator {
+
+    private static final int GET_NPM_VERSION_TIMEOUT_MILLIS = 30_000;
+    private static final int INSTALL_TRACER_TIMEOUT_MILLIS = 300_000;
+
+    @Override
+    public Map<String, String> configure(DatadogTracerJobProperty<?> tracerConfig, Node node, FilePath workspacePath, Map<String, String> envs, TaskListener listener) throws Exception {
+        String nodeVersion = workspacePath.act(new ShellCommandCallable(Collections.emptyMap(), GET_NPM_VERSION_TIMEOUT_MILLIS, "npm", "-v"));
+        listener.getLogger().println("[datadog] Configuring DD JS tracer: got npm version " + nodeVersion + " from " + workspacePath + " on " + node);
+
+        FilePath datadogPath = workspacePath.child(".datadog");
+        datadogPath.mkdirs();
```

Comment:
Do we know if the plugin has permissions to create folders?

Reply:
It has, and we only do things inside the job's workspace